MedallionShell.Tests/AttachingTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Medallion.Shell.Tests
+{
+    [TestClass]
+    public class AttachingTests
+    {
+        [TestMethod]
+        public void TestAttachingToExistingProcess()
+        {
+            var processCommand = Command.Run("SampleCommand", new[] { "sleep", "10000" });
+            var processId = processCommand.ProcessId;
+            Command.TryAttachToProcess(processId, out _)
+                .ShouldEqual(true, "Attaching to process failed.");
+            processCommand.Kill();
+        }
+
+        [TestMethod]
+        public void TestWaitingForAttachedProcessExit()
+        {
+            var processCommand = Command.Run("SampleCommand", new[] { "sleep", "300" });
+            Command.TryAttachToProcess(processCommand.ProcessId, out var attachedCommand)
+                .ShouldEqual(true, "Attaching to process failed.");
+            var commandResult = attachedCommand.Task;
+            commandResult.IsCompleted.ShouldEqual(false, "Task has finished too early.");
+            Thread.Sleep(1000);
```

Comment:
This time will make the test take a whole second to run; a few tests like that can really slow down the suite. Can we flip the timeout above to 100 and this one to 300?

Reply:
Good point, I'll update it.